### PR TITLE
Adding a project to show how mentions works

### PR DESCRIPTION
**Pull request by Virtual-Josh**
This is a sample project to show how mentions works within Teams. This also removes the code about mentions from the Roster project.

### CODE REVIEW

File: tests/Teams/Mentions/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "MicrosoftAppId": "b6b52a0b-88df-455d-8e5c-7169f394f337",
+  "MicrosoftAppPassword": "kZV_rUAL4uQQbC./=WtThzgHKaxOw832"
```
Comment (EricDahlvang) [inline tests/Teams/Mentions/appsettings.json]: These should be empty strings.

File: tests/Teams/Mentions/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+
+            // Create the storage we'll be using for User and Conversation state. (Memory is great for testing purposes.)
+            services.AddSingleton<IStorage, MemoryStorage>();
+
+            // Create the User state. (Used in this bot's Dialog implementation.)
+            services.AddSingleton<ActivityLog>();
```
Comment (johnataylor) [inline tests/Teams/Mentions/Startup.cs]: we don't need the **ActivityLog** in this example

File: tests/Teams/Mentions/ActivityLog.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public class ActivityLog
```
Comment (johnataylor) [inline tests/Teams/Mentions/ActivityLog.cs]: we don't need this class in this sample

File: tests/Teams/Mentions/Bots/MentionsBot.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            var mention = new Mention
+            {
+                Mentioned = turnContext.Activity.From,
+                Text = $"<at>{turnContext.Activity.From.Name}</at>",
+            };
+
+            var replyActivity = MessageFactory.Text($"Hello {mention.Text}.");
+            replyActivity.Entities = new List<Entity> { mention };
+
+            await turnContext.SendActivityAsync(replyActivity);
```
Comment (johnataylor) [inline tests/Teams/Mentions/Bots/MentionsBot.cs]: await turnContext.SendActivityAsync(replyActivity, **cancellationToken**);

File: tests/Teams/Mentions/appsettings.json (hunk 1 above)
Comment (johnataylor) [reply tests/Teams/Mentions/appsettings.json]: +1

**Comment by johnataylor** [comment]
i think it's nice to have this sample - at least it's going to help the docs.

File: tests/Teams/Mentions/appsettings.json (hunk 1 above)
Comment (Virtual-Josh) [reply tests/Teams/Mentions/appsettings.json]: updated.

File: tests/Teams/Mentions/Startup.cs (hunk 2 above)
Comment (Virtual-Josh) [reply tests/Teams/Mentions/Startup.cs]: updated.

File: tests/Teams/Mentions/ActivityLog.cs (hunk 3 above)
Comment (Virtual-Josh) [reply tests/Teams/Mentions/ActivityLog.cs]: removed.

File: tests/Teams/Mentions/Bots/MentionsBot.cs (hunk 4 above)
Comment (Virtual-Josh) [reply tests/Teams/Mentions/Bots/MentionsBot.cs]: added

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

**Pull request merged**